boltdesignsystem/bolt
# Convert components to use custom props

## Jira

[https://pegadigitalit.atlassian.net/browse/DS-36](https://pegadigitalit.atlassian.net/browse/DS-36)
[https://pegadigitalit.atlassian.net/browse/DS-115](https://pegadigitalit.atlassian.net/browse/DS-115)

## Summary

Update the intended web components to use css props instead of sass vars, mix-ins, and functions.

## Details

Mike is working on converting  bolt to css custom props, this PR is to convert all the components' sass files to use these props instead of the vars, mix-ins, and functions already in place. 

Updated components:

- bolt-accordion
- bolt-card
- bolt-carousel
- bolt-code-snippet
- bolt-copy-to-clipboard
- bolt-device-viewer
- bolt-icon
- bolt-image
- bolt-menu
- bolt-modal
- bolt-nav
- bolt-pagination
- bolt-popover
- bolt-progress-bar
- bolt-tabs
- bolt-toc
- bolt-tooltip
- bolt-trigger
- bolt-typeahead
- bolt-video

## How to test

- Open the doc site on both the [build](https://feature-bolt-spacing-css-custom-props-con.boltdesignsystem.com/) and on the live site to make sure both are identical. 
- Check that all the relevant components no longer use the outdated spacing methods (though i imagine the `@warn` will make this easier in time)
